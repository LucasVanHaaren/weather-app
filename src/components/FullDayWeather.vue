<template>
  <v-card class="mx-auto" max-width="600px" outlined>
    <v-list-item>
      <v-list-item-content>
        <div class="headline mb-3">
          <b>{{ name }}</b>
        </div>
        <v-list-item-title class="overline mb-1">
          {{ description}}
        </v-list-item-title>
        <v-list-item>
          <v-list-item-icon>
            <v-icon>fas fa-thermometer-quarter</v-icon>
          </v-list-item-icon>
          <v-list-item-title>
            {{ temperature }} Â°C
          </v-list-item-title>
        </v-list-item>
        <v-list-item>
          <v-list-item-icon>
            <v-icon>fas fa-tint</v-icon>
          </v-list-item-icon>
          <v-list-item-title>
            {{ humidity }} %
          </v-list-item-title>
        </v-list-item>
        <v-list-item>
          <v-list-item-icon>
            <v-icon>fas fa-wind</v-icon>
          </v-list-item-icon>
          <v-list-item-title>
            {{ wind_speed }} m/s
          </v-list-item-title>
        </v-list-item>
        <v-list-item>
          <v-list-item-icon>
            <v-icon>fas fa-tachometer-alt</v-icon>
          </v-list-item-icon>
          <v-list-item-title>
            {{ pressure }} hPa
          </v-list-item-title>
        </v-list-item>
      </v-list-item-content>
      <v-list-item-avatar tile size="200">
        <img :src="icon_url" />
      </v-list-item-avatar>
    </v-list-item>
    <v-card-actions>
      <v-btn width="100%" color="red darken-3" outlined text :to="link">
        See Week Forecast
      </v-btn>
    </v-card-actions>
  </v-card>
</template>

<script>
  export default {
    name: 'FullDayWeather',
    props: {
      name: String,
      description: String,
      temperature: Number,
      humidity: Number,
      pressure: Number,
      wind_speed: Number,
      icon_url: String,
      link: Object
    }
  }
</script>
