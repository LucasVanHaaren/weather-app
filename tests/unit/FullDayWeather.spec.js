import { shallowMount } from '@vue/test-utils'
import FullDayWeather from "@/components/FullDayWeather.vue"

describe('FullDayWeather.vue', () => {
  it('match snapshot', () => {
    const wrapper = shallowMount(FullDayWeather, {
      propsData: {
        name: "Paris",
        description: "Cloudy",
        temperature: 22,
        humidity: 55,
        wind_speed: 54,
        icon_url: "https://openweathermap.org/img/wn/01d@4x.png",
        link: { name: 'WeekWeather', params: { city: "Paris" } }
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
