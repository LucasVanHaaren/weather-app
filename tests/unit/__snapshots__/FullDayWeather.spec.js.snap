// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullDayWeather.vue match snapshot 1`] = `
<v-card max-width="600px" outlined="" class="mx-auto">
  <v-list-item>
    <v-list-item-content>
      <div class="headline mb-3"><b>Paris</b></div>
      <v-list-item-title class="overline mb-1">
        Cloudy
      </v-list-item-title>
      <v-list-item>
        <v-list-item-icon>
          <v-icon>fas fa-thermometer-quarter</v-icon>
        </v-list-item-icon>
        <v-list-item-title>
          22 Â°C
        </v-list-item-title>
      </v-list-item>
      <v-list-item>
        <v-list-item-icon>
          <v-icon>fas fa-tint</v-icon>
        </v-list-item-icon>
        <v-list-item-title>
          55 %
        </v-list-item-title>
      </v-list-item>
      <v-list-item>
        <v-list-item-icon>
          <v-icon>fas fa-wind</v-icon>
        </v-list-item-icon>
        <v-list-item-title>
          54 m/s
        </v-list-item-title>
      </v-list-item>
    </v-list-item-content>
    <v-list-item-avatar tile="" size="200"><img src="https://openweathermap.org/img/wn/01d@4x.png"></v-list-item-avatar>
  </v-list-item>
  <v-card-actions>
    <v-btn width="100%" color="red darken-3" outlined="" text="" to="[object Object]">
      See Week Forecast
    </v-btn>
  </v-card-actions>
</v-card>
`;
